Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIStorage.mm

```diff
@@ -66,12 +71,123 @@
+        @"browser.test.assertThrows(() => browser.storage.session.setAccessLevel({ 'accessLevel': 'INVALID_ACCESS_LEVEL' }), /'accessLevel' value is invalid, because it must specify either 'TRUSTED_CONTEXTS' or 'TRUSTED_AND_UNTRUSTED_CONTEXTS'/i)",
+
         @"browser.test.notifyPass()"
     ]);
 
     Util::loadAndRunExtension(storageManifest, @{ @"background.js": backgroundScript });
 }
 
+TEST(WKWebExtensionAPIStorage, UndefinedProperties)
+{
+    auto *backgroundScript = Util::constructScript(@[
+        @"browser.test.assertEq(typeof browser.storage.local.setAccessLevel, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.sync.setAccessLevel, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.QUOTA_BYTES_PER_ITEM, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.QUOTA_BYTES_PER_ITEM, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.MAX_ITEMS, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.MAX_ITEMS, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.MAX_WRITE_OPERATIONS_PER_HOUR, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.MAX_WRITE_OPERATIONS_PER_HOUR, 'undefined')",
+
+        @"browser.test.assertEq(typeof browser.storage.local.MAX_WRITE_OPERATIONS_PER_MINUTE, 'undefined')",
+        @"browser.test.assertEq(typeof browser.storage.session.MAX_WRITE_OPERATIONS_PER_MINUTE, 'undefined')",
+
+        @"browser.test.notifyPass()"
+    ]);
+
+    Util::loadAndRunExtension(storageManifest, @{ @"background.js": backgroundScript });
+}
+
+TEST(WKWebExtensionAPIStorage, SetAccessLevelTrustedContexts)
+{
+    TestWebKitAPI::HTTPServer server({
+        { "/"_s, { { { "Content-Type"_s, "text/html"_s } }, ""_s } }
+    }, TestWebKitAPI::HTTPServer::Protocol::Http);
+
+    auto *backgroundScript = Util::constructScript(@[
+        @"browser.runtime.onMessage.addListener(async (message, sender, sendResponse) => {",
+        @"  browser.test.assertEq(message, 'Ready')",
+
+        @"  const tabs = await browser.tabs.query({ active: true, currentWindow: true })",
+        @"  const tabId = tabs[0].id",
+
+        @"  await browser.storage.session.setAccessLevel({ accessLevel: 'TRUSTED_CONTEXTS' })",
+
+        @"  const response = await browser.test.assertSafeResolve(() => browser.tabs.sendMessage(tabId, { content: 'Access level set' }))",
+        @"  browser.test.assertEq(response?.content, 'undefined')",
+
+        @"  browser.test.notifyPass()",
+        @"})"
+    ]);
+
+    auto *contentScript = Util::constructScript(@[
+        @"browser.runtime.onMessage.addListener((message, sender, sendResponse) => {",
+        @"  browser.test.assertEq(message.content, 'Access level set', 'Should receive the correct message content')",
```

Comment:
Just `message` if you make the other change in the `sendMessage` part. Otherwise `message?.content` to avoid breaking if the message is a different object.